src/bio/bio_buffer.c

```diff
@@ -802,7 +812,8 @@ bio_memcpy(struct bio_desc *biod, uint16_t media, void *media_addr,
 {
 	struct umem_instance *umem = biod->bd_ctxt->bic_umem;
 
-	if (biod->bd_update && media == DAOS_MEDIA_SCM) {
+	D_ASSERT(biod->bd_type < BIO_IOD_TYPE_GETBUF);
```

Comment:
same comment here

Reply:
Sure, I'll add more comments in follow-on BIO changes.